### PR TITLE
Add MNK50

### PR DESCRIPTION
**Pull request by yiancar**
## Description

Add MNK50 keyboard
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mnk50/rules.mk [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = STM32F072
+
+# Do not put the microcontroller into power saving mode
```
Comment (drashna) [inline keyboards/mnk50/rules.mk:4]:
```suggestion
# Bootloader selection
BOOTLOADER = stm32-dfu

# Do not put the microcontroller into power saving mode
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/mnk50/rules.mk [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+NO_USB_STARTUP_CHECK = no   # Disable initialization only when usb is plugged in
```
Comment (fauxpark) [inline keyboards/mnk50/rules.mk:28]:
```suggestion
```

File: keyboards/mnk50/readme.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+* Hardware Supported: A 40% keyboard with STM32F072CB
+* Hardware Availability: https://www.monokei.co
+
+## Instructions
+
+### Build
+
+Make example for this keyboard (after setting up your build environment):
+
+    make mnK50:default
```
Comment (fauxpark) [inline keyboards/mnk50/readme.md:15]:
```suggestion
    make mnk50:default
```

File: keyboards/mnk50/keymaps/default/readme.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# The default keymap for MNK50. VIA support disabled.
```
Comment (fauxpark) [inline keyboards/mnk50/keymaps/default/readme.md:1]:
```suggestion
# The default keymap for MNK50
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by yiancar** [comment]
I think we good now?:)

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**